drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1203,7 +1203,11 @@ static int eth_initialize(const struct device *dev)
 	memset(&tx_config, 0, sizeof(ETH_TxPacketConfig));
 	tx_config.Attributes = ETH_TX_PACKETS_FEATURES_CSUM |
 				ETH_TX_PACKETS_FEATURES_CRCPAD;
+#if defined(CONFIG_ETH_STM32_HW_CHECKSUM)
 	tx_config.ChecksumCtrl = ETH_CHECKSUM_IPHDR_PAYLOAD_INSERT_PHDR_CALC;
+#else
+	tx_config.ChecksumCtrl = ETH_CHECKSUM_DISABLE;
+#endif
```

Comment:
For readability, as this code section is already cluttered with #ifdef's, I'd suggest: 
```
	tx_config.ChecksumCtrl = IS_ENABLED(CONFIG_ETH_STM32_HW_CHECKSUM) ?
				ETH_CHECKSUM_IPHDR_PAYLOAD_INSERT_PHDR_CALC : ETH_CHECKSUM_DISABLE;
```